### PR TITLE
[var] Fix hb_ot_var_get_axis_infos's offset semantic

### PR DESCRIPTION
**Pull request by ebraminio**
The API was adding offset to input's infos buffer index also which is
unusual between our APIs and wrong.

### CODE REVIEW

File: src/hb-ot-var-fvar-table.hh [hunk 1]
```diff
@@ -173,17 +164,8 @@ struct fvar
   {
     if (axes_count)
     {
-      /* TODO Rewrite as hb_array_t<>::sub-array() */
-      unsigned int count = axisCount;
-      start_offset = hb_min (start_offset, count);
-
-      count -= start_offset;
-      axes_array += start_offset;
```
Comment (ebraminio) [inline src/hb-ot-var-fvar-table.hh]: This is the part that was wrong, start_offset isn't intended to be added to axes_array index but just used just as a AxisRecord index offset.

File: test/api/test-var-coords.c [hunk 2]
```diff
@@ -67,10 +67,29 @@ test_get_var_coords (void)
...
+
+  unsigned num_coords = hb_ot_var_get_axis_count (face);
+  hb_ot_var_axis_info_t info;
+  unsigned c = 1;
+
+  hb_ot_var_get_axis_infos (face, 0, &c, &info);
+  g_assert (info.tag == HB_TAG ('w','g','h','t'));
+
+  hb_ot_var_get_axis_infos (face, 1, &c, &info);
+  g_assert (info.tag == HB_TAG ('w','d','t','h'));
```
Comment (ebraminio) [inline test/api/test-var-coords.c]: This didn't work…

**Comment by ebraminio** [comment]
@khaledhosny @behdad What do you suggest? Do you know clients that were using offset of the API? This overflows buffer for clients who iterating axis in usual way of HarfBuzz API.

**Comment by ebraminio** [comment]
Pango fortunately doesn't use its offset, guess no client would use as it is broken anyway.

**Comment by ebraminio** [comment]
Any pointers?

**Comment by behdad** [comment]
Yeah I think is safe to fix.

No idea why I didn't just use sub_array to begin with...  But yeah I put TODOs where I insert bugs apparently!

**Comment by ebraminio** [comment]
Thank you :)

**Pull request merged**